script.js

```diff
@@ -0,0 +1,91 @@
+
+const resetButton = document.querySelector('button[data-behaviour="reset"]');
+const sizeButton = document.querySelector('button[data-behaviour="size"]');
+const rainbowButton = document.querySelector('button[data-behaviour="rainbow"]');
+const blackButton = document.querySelector('button[data-behaviour="black"]');
+const gameContainer = document.querySelector('.gameContainer');
+let isRainbowPen = false
+sizeButton.addEventListener('click', () => {
+  makeGrid()
+});
+
+rainbowButton.addEventListener('click', () => {
+  isRainbowPen = true
+});
+
+blackButton.addEventListener('click', () => {
+  isRainbowPen = false
+});
+
+
+
+//Create Grid
+function grid(size) {
+  for(let i = 1; i <= (size * size); i++) {
+    let gridCell = document.createElement('div');
+    gridCell.classList.add('grid');
+    gameContainer.appendChild(gridCell);
+    document.querySelectorAll('grid');
+    gridCell.addEventListener('mouseover', () => colorGridCell(gridCell));
+    gridCell.addEventListener('click', () => gridCell.classList.remove('paint'));
+  }
+}
+grid(16);
+
+//Resets the game and prompts the user for a numerical input for the size of the new grid.
+function makeGrid() {
+  let gridSize = prompt('How many squares would you like on each axis of the new grid?', 16, )
+  document.body.style.setProperty('--gridSize', gridSize)
+  let elementRemove = document.getElementsByClassName('grid')
+  while (elementRemove[0]) {
```

Comment:
Can you find a more performant way of doing this? Loops will always be more expensive and only used when necessary